### PR TITLE
PMON Voltage/Current Sensor Monitoring Enhancement

### PR DESCRIPTION
**Pull request by bmridul**
HLD for Voltage and Current sensor monitoring.

Related PRs:

| PR | Description | Status |
|----|------------|--------|
|https://github.com/sonic-net/sonic-utilities/pull/2941 | Sensormon CLIs | Merged |
|https://github.com/sonic-net/sonic-platform-common/pull/393 | Sensors Base Classes | Merged|
|https://github.com/sonic-net/sonic-platform-daemons/pull/393 | Sensormon Daemon | Merged|
|https://github.com/sonic-net/sonic-buildimage/pull/16089 | Add Sensormon in PMON | Merged|
|https://github.com/sonic-net/sonic-buildimage/pull/16646 | System Health support | Open|
|https://github.com/sonic-net/sonic-snmpagent/pull/295 | SNMP support | Open|
|https://github.com/sonic-net/sonic-mgmt/pull/24036 | SONiC Mgmt test cases | Merged|
|https://github.com/sonic-net/sonic-platform-common/pull/426 | File system bases sensors | Merged|

### CODE REVIEW

**Comment by prgeor** [comment]
@bmridul why not use the existing `lm-sensors` in linux which provides much more detailed output ? We have `sensord` running in PMON which can monitor voltage/current. We also syslog alerts for high temperature and current.

```
root@3POs-8101-H11-18:~# docker exec pmon service sensord status
sensord is running.
root@3POs-8101-H11-18:~#
```

File: doc/pmon/pmon-sensormon.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+Note that temperature sensor devices are managed via SONiC ThermalCtlD daemon today. At this point there is no change proposed for ThermalCtlD. This proposed design can be used for voltage, current and other types of sensors.
+
+
+### Requirements
+
+This HLD covers
+
+* Discovery of voltage and current sensor devices in the system
+* Monitoring the sensor devices periodically and update that data in Redis DB
+* Raise/Clear alarms if the sensor devices indicate readings which are unexpected and clear them if they return to a good state.
```
Comment (prgeor) [inline doc/pmon/pmon-sensormon.md:47]: @bmridul aren't these already done by `sensord`

File: doc/pmon/pmon-sensormon.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+The proposal for monitoring sensor devices is to create a new SensorMon daemon. SensorMon will use API provided by platform to discover the sensor devices. It will periodically poll the devices to refresh the sensor information and update the readings in StateDB.
+
+If the sensor device readings cross the minor/major/critical thresholds, syslogs will be generated to indicate to the operator about the alarm condition. If the sensor reports normal data in a subsequent poll, another syslog will be generated to indicate that the fault condition is cleared.
+
+CLIs are provided to display the sensor devices, their measurements, threshold values and if they are reporting an alarm.
+
+Platform APIs will provide 
+
+* List of sensors devices of a specific type in the system
+* Way to read the sensor information
```
Comment (prgeor) [inline doc/pmon/pmon-sensormon.md:66]: @bmridul can we make use of exisiting `sensors` command ?

File: doc/pmon/pmon-sensormon.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+The following SONiC repositories will have changes
+
+#### sonic-platform-daemons	
+
+SensorMon will be a new daemon that will run in PMON container. It will retrieve a list of sensors of different sensor types from the platform during initialization. Subsequently, it will poll the sensor devices on a periodic basis and update their measurments in StateDb. SemsorMon will also raise syslogs on alarm conditions.
```
Comment (prgeor) [inline doc/pmon/pmon-sensormon.md:72]: @bmridul sensord already does all these including raising ALERT
```
Nov  3 19:03:00.468898 sonic ALERT pmon#sensord: Sensor alarm: Chip dps460-i2c-10-59: PSU-1(L) 220V Rail (in): -589824.00 V (min = -0.50 V, max = -0.50 V) [ALARM]
Nov  3 19:03:00.469262 sonic ALERT pmon#sensord: Sensor alarm: Chip dps460-i2c-10-59: vout1: +536608768.00 V (min = +1071628288.00 V, max = +1071628288.00 V) [ALARM]
Nov  3 19:03:00.469388 sonic ALERT pmon#sensord: Sensor alarm: Chip dps460-i2c-10-59: PSU-1(L) Fan 1: -1 RPM [ALARM]
```

**Comment by bmridul** [comment]
> @bmridul why not use the existing `lm-sensors` in linux which provides much more detailed output ? We have `sensord` running in PMON which can monitor voltage/current. We also syslog alerts for high temperature and current.
> 
> ```
> root@3POs-8101-H11-18:~# docker exec pmon service sensord status
> sensord is running.
> root@3POs-8101-H11-18:~#
> ```

Hi @prgeor, 

Please note that the solution proposed is not mutually exclusive to Sensord/Lmsensors. The solution proposed here can be used alongside or without Sensord.

There are 2 main reasons we are proposing this. 

1.	Sensord provides monitoring of sensors which are supported by the linux hwmon infrastructure. The list of devices supported are mentioned here : https://hwmon.wiki.kernel.org/device_support_status

However, the issue is that there are a number of devices which report voltage/current measurements that are not supported by hwmon.

An example is device LTC2497 which can report voltage measurements.

The open source driver available for this device supports a different linux infrastructure – Industrial I/O
https://www.kernel.org/doc/html/v4.12/driver-api/iio/intro.html

The driver code is here : https://github.com/torvalds/linux/blob/master/drivers/iio/adc/ltc2497.c

This device will not be monitored by hwmon unless we patch its standard kernel driver which is not something we would want to do.

Sensormon can cover this gap and provide one CLI/rpcoess to view/monitor all the voltage and current sensors.

2.	Even when hwmon does support the devices, we observed that the fault reporting only works when the device indicates a fault. A user supplied threshold would not generate a minor or major alarm. Sensormon can provide this support and alert if there is a failure even when the device may not have the capability to generate an alarm.

File: doc/pmon/pmon-sensormon.md (hunk 2 above)
Comment (bmridul) [reply doc/pmon/pmon-sensormon.md:66]: As mentioned in the other comment, you can use sensors for the hwmon supported sensors if platform vendor prefers that. Sensors.conf provides flexibility if you want the sensor to be monitored by sensord or not.

File: doc/pmon/pmon-sensormon.md (hunk 3 above)
Comment (bmridul) [reply doc/pmon/pmon-sensormon.md:72]: Sensord does raise alarm if the device reports an alarm.
https://stuff.mit.edu/afs/sipb/user/dmaze/debian/Packages/lm-sensors/lm-sensors-2.8.0/doc/lm_sensors-FAQ.html#Section%204.4
However certain hardware designs will recommend monitoring voltage readings to generate alarms (even though device may not be reporting or not be capable of reporting an alarm).

File: doc/pmon/pmon-sensormon.md (hunk 1 above)
Comment (bmridul) [reply doc/pmon/pmon-sensormon.md:47]: Answered in previous comment.

**Comment by prgeor** [comment]
> > @bmridul why not use the existing `lm-sensors` in linux which provides much more detailed output ? We have `sensord` running in PMON which can monitor voltage/current. We also syslog alerts for high temperature and current.
> > ```
> > root@3POs-8101-H11-18:~# docker exec pmon service sensord status
> > sensord is running.
> > root@3POs-8101-H11-18:~#
> > ```
> 
> Hi @prgeor,
> 
> Please note that the solution proposed is not mutually exclusive to Sensord/Lmsensors. The solution proposed here can be used alongside or without Sensord.
> 
> There are 2 main reasons we are proposing this.
> 
> 1. Sensord provides monitoring of sensors which are supported by the linux hwmon infrastructure. The list of devices supported are mentioned here : https://hwmon.wiki.kernel.org/device_support_status
> 
> However, the issue is that there are a number of devices which report voltage/current measurements that are not supported by hwmon.
> 
> An example is device LTC2497 which can report voltage measurements.
> 
> The open source driver available for this device supports a different linux infrastructure – Industrial I/O https://www.kernel.org/doc/html/v4.12/driver-api/iio/intro.html
> 
> The driver code is here : https://github.com/torvalds/linux/blob/master/drivers/iio/adc/ltc2497.c
> 
> This device will not be monitored by hwmon unless we patch its standard kernel driver which is not something we would want to do.
> 
> Sensormon can cover this gap and provide one CLI/rpcoess to view/monitor all the voltage and current sensors.
> 
> 2. Even when hwmon does support the devices, we observed that the fault reporting only works when the device indicates a fault. A user supplied threshold would not generate a minor or major alarm. Sensormon can provide this support and alert if there is a failure even when the device may not have the capability to generate an alarm.

@bmridul I am still not quite convinced the use of sensormon, since devices using IIO is mostly used in consumer products like laptop, mobile phones. The operating range of the chip is best known to the chip manufacturer and hence its better to depend upon the chip based alarm rather than platform integrator arriving at the customized threshold to raise alarms. As you rightly pointed out, hwmon already includes almost all sensors barring few exceptions, so I would prefer if you make effort to include these few exception devices included in hwmon support.

Nevertheless, please capture the reasons in the HLD for sensormon.

**Comment by bmridul** [comment]
> > > @bmridul why not use the existing `lm-sensors` in linux which provides much more detailed output ? We have `sensord` running in PMON which can monitor voltage/current. We also syslog alerts for high temperature and current.
> > > ```
> > > root@3POs-8101-H11-18:~# docker exec pmon service sensord status
> > > sensord is running.
> > > root@3POs-8101-H11-18:~#
> > > ```
> > 
> > 
> > Hi @prgeor,
> > Please note that the solution proposed is not mutually exclusive to Sensord/Lmsensors. The solution proposed here can be used alongside or without Sensord.
> > There are 2 main reasons we are proposing this.
> > 
> > 1. Sensord provides monitoring of sensors which are supported by the linux hwmon infrastructure. The list of devices supported are mentioned here : https://hwmon.wiki.kernel.org/device_support_status
> > 
> > However, the issue is that there are a number of devices which report voltage/current measurements that are not supported by hwmon.
> > An example is device LTC2497 which can report voltage measurements.
> > The open source driver available for this device supports a different linux infrastructure – Industrial I/O https://www.kernel.org/doc/html/v4.12/driver-api/iio/intro.html
> > The driver code is here : https://github.com/torvalds/linux/blob/master/drivers/iio/adc/ltc2497.c
> > This device will not be monitored by hwmon unless we patch its standard kernel driver which is not something we would want to do.
> > Sensormon can cover this gap and provide one CLI/rpcoess to view/monitor all the voltage and current sensors.
> > 
> > 2. Even when hwmon does support the devices, we observed that the fault reporting only works when the device indicates a fault. A user supplied threshold would not generate a minor or major alarm. Sensormon can provide this support and alert if there is a failure even when the device may not have the capability to generate an alarm.
> 
> @bmridul I am still not quite convinced the use of sensormon, since devices using IIO is mostly used in consumer products like laptop, mobile phones. The operating range of the chip is best known to the chip manufacturer and hence its better to depend upon the chip based alarm rather than platform integrator arriving at the customized threshold to raise alarms. As you rightly pointed out, hwmon already includes almost all sensors barring few exceptions, so I would prefer if you make effort to include these few exception devices included in hwmon support.
> 
> Nevertheless, please capture the reasons in the HLD for sensormon.

@prgeor , Thanks for your comments. I will update the HLD with more details. Few responses

1. The component vendors might push a driver based on initial application of the device. However the components can be used in a variety of applications once generally available. 
2. Many devices do not provide an alarm function. If the device has hardware logic for monitoring the voltage/current, it might provide an alarm, but many devices do not.
3. Even though the device might be in its max/min limits, it might have an adverse affect on the components it is delivering the voltage to. So the sensor thresholds should be adjusted from a board design perspective. 

The uber point is that the sensord may not provide complete coverage. If Linux Sensord is good enough for a platform, it can choose not to use this infra. We can discuss further in community meeting.

**Comment by jeff-yin** [comment]
As discussed during the community call today -- please explore integration with SONiC PDDF (https://github.com/sonic-net/SONiC/blob/master/doc/platform/brcm_pdk_pddf.md) so that platform vendors can easily integrate support for sensors using their PDDF JSON files.

**Review by Junchao-Mellanox — CHANGES_REQUESTED** [review]
A few general comments:
1. How is the daemon connect to system health service?
2. How is the daemon connect to snmp?
3. How to build the sensor hierarchy relation? For example, should PSU voltage sensor object under PSU object or under chassis object?
4. There is already logic in PSUD to handle PSU voltage threshold, do you plan to remove it?
5. How to handle hot swappable sensors? For example, PSU sensor, module sensor?

**Comment by bmridul** [comment]
> As discussed during the community call today -- please explore integration with SONiC PDDF (https://github.com/sonic-net/SONiC/blob/master/doc/platform/brcm_pdk_pddf.md) so that platform vendors can easily integrate support for voltage/current sensors using their PDDF JSON files.

I looked briefly into the PDDF documentation and some code. It seems the PDDF support for voltage and current sensors can be added in a similar way as as other platform components are supported. E,g, The code below is implementing the platform support for thermal sensors.

https://github.com/sonic-net/sonic-buildimage/blob/204579a0ccfe80c1f2afea2bda0574528faecb91/platform/pddf/platform-api-pddf-base/sonic_platform_pddf_base/pddf_thermal.py

Voltage and current sensors support can be similarly added.

**Comment by Junchao-Mellanox** [comment]
> A few general comments:
> 
> 1. How is the daemon connect to system health service?
> 2. How is the daemon connect to snmp?
> 3. How to build the sensor hierarchy relation? For example, should PSU voltage sensor object under PSU object or under chassis object?
> 4. There is already logic in PSUD to handle PSU voltage threshold, do you plan to remove it?
> 5. How to handle hot swappable sensors? For example, PSU sensor, module sensor?

Hi @bmridul @abdosi , how about these comments?

**Comment by bmridul** [comment]
> A few general comments:
> 
> 1. How is the daemon connect to system health service?

It would be good to show the out of out of range sensor readings in system health o/p. Currently out of range temperature sensors are also not shown in system health. We should add those as well. 

I will update HLD and open a PR for this. I believe this can be added here
https://github.com/sonic-net/sonic-buildimage/blob/master/src/system-health/health_checker/hardware_checker.py

**Comment by bmridul** [comment]
> A few general comments:
> 2. How is the daemon connect to snmp?

SNMP support will need to be added to represent the voltage/current sensors in Entity MIB. We will need to address this with changes in rfc2737 implmentation.

**Comment by bmridul** [comment]
> A few general comments:
> 3. How to build the sensor hierarchy relation? For example, should PSU voltage sensor object under PSU object or under chassis object?

Currently this HLD is proposing support for voltage and current sensors on the system boards (RPs/LCs/System Board). PSUs are already handled by PSUd so we are not proposing to change the PSUd implementation.
 
> 4. There is already logic in PSUD to handle PSU voltage threshold, do you plan to remove it?
No, Pls see above.

> 5. How to handle hot swappable sensors? For example, PSU sensor, module sensor?
The implementation does cover the case if a sensor becomes unavailable and a message will be logged if the sensor is removed from the system.

**Comment by bmridul** [comment]
HLD updated with review comments.

**Comment by abdosi** [comment]
@bmridul Please capture in HLD what features/implementation are in phase-1 already for which PR has been done and merged. And what will be done in phase-2 (not immediate)

**Comment by bmridul** [comment]
> @bmridul Please capture in HLD what features/implementation are in phase-1 already for which PR has been done and merged. And what will be done in phase-2 (not immediate)

Added.

**Comment by bmridul** [comment]
> > A few general comments:
> > 2. How is the daemon connect to snmp?
> 
> SNMP support will need to be added to represent the voltage/current sensors in Entity MIB. We will need to address this with changes in rfc2737 implmentation.

SNMP support PR added.

**Comment by bmridul** [comment]
@Junchao-Mellanox @jeff-yin @prgeor , Pls check if ur comments have been addressed.

**Comment by jeff-yin** [comment]
> @Junchao-Mellanox @jeff-yin @prgeor , Pls check if ur comments have been addressed.

I didn't really see any updates with regards to PDDF specifically. If it's not going to be covered, please indicate that in the HLD and provide a path for integration in the future.

**Comment by bmridul** [comment]
> > @Junchao-Mellanox @jeff-yin @prgeor , Pls check if ur comments have been addressed.
> 
> I didn't really see any updates with regards to PDDF specifically. If it's not going to be covered, please indicate that in the HLD and provide a path for integration in the future.

Hi Jeff, 

I had replied to your comment above - https://github.com/sonic-net/SONiC/pull/1394#issuecomment-1711136593

Currently our platforms donot support PDDF infra so I am not sure how to test PDDF integration. However it seems straightforward to me. I can open a git issue for this and future integration can be tracked that way. LMK.

File: doc/pmon/pmon-sensormon.md (hunk 3 above)
Comment (prgeor) [reply doc/pmon/pmon-sensormon.md:82]: @bmridul can we specify customer threshold in sensors conf file

https://www.kernel.org/doc/Documentation/hwmon/sysfs-interface


image

Comment (prgeor) [reply doc/pmon/pmon-sensormon.md:82]: image

**Comment by jeff-yin** [comment]
> > > @Junchao-Mellanox @jeff-yin @prgeor , Pls check if ur comments have been addressed.
> 
> > 
> 
> > I didn't really see any updates with regards to PDDF specifically. If it's not going to be covered, please indicate that in the HLD and provide a path for integration in the future.
> 
> 
> 
> Hi Jeff, 
> 
> 
> 
> I had replied to your comment above - https://github.com/sonic-net/SONiC/pull/1394#issuecomment-1707697964
> 
> 
> 
> Currently our platforms donot support PDDF infra so I am not sure how to test PDDF integration. However it seems straightforward to me. I can open a git issue for this and future integration can be tracked that way. LMK.

Sure, that's fine. Please do create the issue to track it.

**Review by jeffyin-dell — DISMISSED** [review]

File: doc/pmon/pmon-sensormon.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+Note that temperature sensor devices are managed via SONiC ThermalCtlD daemon today. At this point there is no change proposed for ThermalCtlD. This proposed design can be used for voltage, current and other types of sensors.
+
+Linux does provide some support of voltage and current sensor monitoring using lmsensors/hwmon infrastructure. However there are a few limitations with that
+
+- Devices not supported with Hwmon are not covered
```
Comment (prgeor) [inline doc/pmon/pmon-sensormon.md:42]: @bmridul could you list few such devices not supported by hwmon? Why not we add support for them in hwmon?

Comment (bmridul) [reply doc/pmon/pmon-sensormon.md:42]: I donot have a comprehensive list of devices. I had looked at the hardware designs in our company and found some devices for which the standard drivers did not include hwmon support. This is what I replied in the [comment](https://github.com/sonic-net/SONiC/pull/1394#issuecomment-1638528187) earlier.  A bunch of devices seem to support Linux IIO infra and donot include the hwmon/lmsensors support.

https://github.com/torvalds/linux/tree/master/drivers/iio/adc
https://github.com/torvalds/linux/tree/master/drivers/iio/dac

These drivers are contributed from the device vendors and adding hwmon support will need us to co-ordinate with different vendors and would take a long time. 
.

File: doc/pmon/pmon-sensormon.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+
+The following SONiC repositories will have changes
+
+#### sonic-platform-daemons	
+
+SensorMon will be a new daemon that will run in PMON container. It will retrieve a list of sensors of different sensor types from the platform during initialization. Subsequently, it will poll the sensor devices on a periodic basis and update their measurments in StateDb. SensorMon will also raise syslogs on alarm conditions. 
+
+Following is the DB schema for voltage and current sensor data.
+
+##### Voltage Sensor StateDb Schema
```
Comment (prgeor) [inline doc/pmon/pmon-sensormon.md:86]: @bmridul in which format platform specify different onboard sensors and respective threshold Can you make the format public?

Comment (bmridul) [reply doc/pmon/pmon-sensormon.md:86]: @prgeor , Pls check 
https://github.com/sonic-net/SONiC/blob/3ed3cca2866e3647b8af9871b33dc53b5dbae1a6/doc/pmon/pmon-sensormon.md#platform-sensors-configuration

Comment (prgeor) [reply doc/pmon/pmon-sensormon.md:86]: @bmridul so this will work only for platforms having sysfs for each sensors?

Comment (bmridul) [reply doc/pmon/pmon-sensormon.md:86]: Yes,  For platforms having sysfs based sensors, they only need to provide sensors details in this file and Sensormon will use that.

File: doc/pmon/pmon-sensormon.md [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+Platform APIs will provide 
+
+* List of sensors devices of a specific type in the system
+* Way to read the sensor information
+
+The following SONiC repositories will have changes
+
+#### sonic-platform-daemons	
+
+SensorMon will be a new daemon that will run in PMON container. It will retrieve a list of sensors of different sensor types from the platform during initialization. Subsequently, it will poll the sensor devices on a periodic basis and update their measurments in StateDb. SensorMon will also raise syslogs on alarm conditions.
```
Comment (prgeor) [inline doc/pmon/pmon-sensormon.md:82]: @bmridul Which APIs are you referring to here? Can you point me to the platform common APIs?

Comment (bmridul) [reply doc/pmon/pmon-sensormon.md:82]: @prgeor , Pls check the following.

https://github.com/sonic-net/sonic-platform-common/blob/master/sonic_platform_base/sensor_base.py
https://github.com/sonic-net/sonic-platform-common/blob/c82ae543642411676fe9afa2bac32f733356cfc7/sonic_platform_base/chassis_base.py#L457

File: doc/pmon/pmon-sensormon.md [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+    	...
+    	VP0P75_CORE_NPU2            Not OK    voltage
+    	...
+
+
+##### Platform Sensors Configuration
+
+Sensormond will use the platform APIs for retrieving platform sensor information. However, for platforms with only file-system/sysfs based drivers, a simple implementation is provided wherein the platform can specify the sensor information for the board and any submodules (such as fabric cards) in a data file and Sensormond can use that for finding sensors and monitoring them. 
+
+The file system/Sysfs based platform sensor information can be provided using a yaml file. The yaml file shall have the following format.
```
Comment (prgeor) [inline doc/pmon/pmon-sensormon.md:200]: @bmridul what is the location of the yaml file for platform? Is at HWSKU level or platform level?

Comment (bmridul) [reply doc/pmon/pmon-sensormon.md:200]: Its at the platform level -  /usr/local/etc/sensors.yaml
Pls see https://github.com/sonic-net/sonic-platform-common/pull/426

File: doc/pmon/pmon-sensormon.md [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+	POL_CORE_N0_I2       22250 mA      30000     18000           28000          15000      False  20230212 11:18:28
+
+
+	
+
+#### Configuration and Management
+
+At this point, there is no configuration requirement for this feature. 
+
+If the SensorMon daemon is not desired to be run in the system, an entry can be added to  pmon_daemon_control.json to exclude it from running in the system.
```
Comment (prgeor) [inline doc/pmon/pmon-sensormon.md:171]: @bmridul can we ensure that existing platform don't have to explicitly follow this i.e if sensor.ymal is not present then sensormon does not run?

File: doc/pmon/pmon-sensormon.md [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+            high_thresholds: [ <critical>, <major>, <minor> ]
+            low_thresholds: [ <critical>, <major>, <minor> ]
+          ...
+
+
+
+
+##### PDDF Support
+
+SONiC PDDF provides a data driven framework to access platform HW devices. PDDF allows for sensor access information to be read from platform specific json files. PDDF support can be added for voltage and current sensors which can be retrieved by Sensormon.
```
Comment (prgeor) [inline doc/pmon/pmon-sensormon.md:239]: @bmridul in this HLD scope, are you including to parse the .json too if a platform choose PDDF approach?Why can't we use existing PDDF approach to read from .json?

Comment (bmridul) [reply doc/pmon/pmon-sensormon.md:239]: You can use  API OR Yaml file OR PDDF  approach. Sensormon will work with any of these approaches used for platform support.

File: doc/pmon/pmon-sensormon.md (hunk 8 above)
Comment (bmridul) [reply doc/pmon/pmon-sensormon.md:171]: If platform does not provide the sensor definition, then there is no impact. Sensormond will not do anything. The pmon_daemon_control.json is simply if the platform does not want to have the daemon running. This is same approach used for other PMON daemons (e.g. Same approach is for PSUd - https://github.com/sonic-net/sonic-buildimage/blob/927dde73f12629f3a4da84a729fa6ea65cad9676/dockers/docker-platform-monitor/docker-pmon.supervisord.conf.j2#L134)

**Review by prgeor — APPROVED** [review]

**Comment by zhangyanzhao** [comment]
HLD is merged but code PRs are still open, move to backlog

**Pull request merged**